moveit_ros/planning/planning_request_adapter_plugins/src/fix_start_state_collision.cpp

```diff
@@ -115,37 +126,86 @@ class FixStartStateCollision : public planning_request_adapter::PlanningRequestA
       planning_scene->checkCollision(vcreq, vcres, start_state);
 
       if (creq.group_name.empty())
-        ROS_INFO("Start state appears to be in collision");
+        ROS_INFO("Start state appears to be in collision. Attempting to fix.");
       else
-        ROS_INFO_STREAM("Start state appears to be in collision with respect to group " << creq.group_name);
+        ROS_INFO_STREAM("Start state appears to be in collision with respect to group " << creq.group_name
+                                                                                        << ". Attempting to fix.");
 
-      moveit::core::RobotStatePtr prefix_state(new moveit::core::RobotState(start_state));
-      random_numbers::RandomNumberGenerator& rng = prefix_state->getRandomNumberGenerator();
+      moveit::core::RobotStatePtr original_state(new moveit::core::RobotState(start_state));
+      random_numbers::RandomNumberGenerator& rng = original_state->getRandomNumberGenerator();
 
       const std::vector<const moveit::core::JointModel*>& jmodels =
           planning_scene->getRobotModel()->hasJointModelGroup(req.group_name) ?
               planning_scene->getRobotModel()->getJointModelGroup(req.group_name)->getJointModels() :
               planning_scene->getRobotModel()->getJointModels();
 
+      // Record cartesian pose of every link to ensure no sudden movements will occur
+      std::vector<const moveit::core::LinkModel*> link_models;
+      link_models.resize(jmodels.size());
+      for (std::size_t i = 0; i < jmodels.size(); ++i)
+        link_models[i] = jmodels[i]->getChildLinkModel();
+
+      // Also record end effector link pose if one is set
+      if (!req.goal_constraints[0].position_constraints.empty())
+        link_models.push_back(start_state.getLinkModel(req.goal_constraints[0].position_constraints[0].link_name));
+      else if (!req.goal_constraints[0].orientation_constraints.empty())
+        link_models.push_back(start_state.getLinkModel(req.goal_constraints[0].orientation_constraints[0].link_name));
```

Comment:
Nice catch and good call on reversing the link order. It's only a max of two since it only adds the link from the position _or_ the orientation constraint, but it's still redundant. I added a solution that isn't very elegant, but probably still better than wasting calculations.